src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -325,10 +327,53 @@ internal static bool ValidateParentTagName(string parentTag, ErrorSink errorSink
                     errorSink: errorSink);
         }
 
-        private static bool ValidateName(
-            string name,
-            bool targetingAttributes,
-            ErrorSink errorSink)
+        private static bool ValidateRequiredAttributeName(string attributeName, ErrorSink errorSink)
+        {
+            if (attributeName.Length >= 1 && attributeName[0] == '[')
+            {
+                // Value starts with '[', treat it as a CSS selector.
+
+                if (attributeName[attributeName.Length - 1] != ']')
+                {
+                    errorSink.OnError(SourceLocation.Zero, "TODO: Create error for invalid CSS Selector syntax", length: 0);
+                    return false;
+                }
+
+                var equalsIndex = attributeName.IndexOf('=');
+                if (equalsIndex != -1)
+                {
+                    if (attributeName.IndexOfAny(UnsupportedAttributeSelectors) == equalsIndex - 1)
+                    {
+                        errorSink.OnError(SourceLocation.Zero, "TODO: Create error for unsupported CSS Selector.", length: 0);
+                        return false;
+                    }
+
+                    // Don't need to bounds check here because the ']' error validation ensures that there's at least 1
+                    // extra character after the '='.
+                    var afterEquals = attributeName[equalsIndex + 1];
+                    if ((afterEquals == '\'' || afterEquals == '"') &&
+                        attributeName[attributeName.Length - 2] != afterEquals)
+                    {
+                        errorSink.OnError(SourceLocation.Zero, "TODO: Create error for unbalanced quotes for CSS Selector.", length: 0);
+                        return false;
+                    };
+                }
```

Comment:
Where is error for case where value does not contain an attribute name e.g. `[='something']` or `[$="something"]`?